documentation/boltspec_reference.md

```diff
@@ -0,0 +1,1117 @@
+# BoltSpec reference
+
+The `BoltSpec` helper library includes several functions to help write unit
+tests for plans. For more information about writing unit tests for plans, and
+how to use these functions, see [Testing plans](testing_plans.md).
+
+## Mocks
+
+Mocks serve two purposes when you are writing tests for your plans: they allow
+assertions about executor functions invoked during a plan run and also allow you
+to set their return values. For example, you might use a mock to make an
```

Comment:
```suggestion
to mock their return values. For example, you might use a mock to make an
```

Reply:
idk if this is just more confusing, like using a word in it's own definition :P

Reply:
I think using mock in two contexts here could be confusing, so I'll leave it as 'set'.